### PR TITLE
Adding non-IDE build framework

### PR DESCRIPTION
**Pull request by bbaltz505**
Build examples without IDE. Requires:
* CODK_DIR environment variable to be set
* arc-elf32-xxx commands in your path (new toolchain)

### CODE REVIEW

File: bin/x86/bin/install_script.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+# fix dynamic loader paths in all ELF SDK binaries
+native_sysroot=$($SUDO_EXEC cat "$env_setup_script" |grep OECORE_NATIVE_SYSROOT|cut -d'=' -f2|tr -d '"')
+dl_path=$($SUDO_EXEC find "$native_sysroot"/lib -name "ld-linux*")
+if [ "$dl_path" = "" ] ; then
+	echo "SDK could not be set up. Relocate script unable to find ld-linux.so. Abort!"
+	exit 1
+fi
+#executable_files="'"$($SUDO_EXEC find "$native_sysroot" -type f -perm +111)"'"
+executable_files=$($SUDO_EXEC find "$native_sysroot" -type f -perm +111 -exec printf "\"%s\" " {} \; )
```
Comment (eriknyquist) [inline bin/x86/bin/install_script.sh]: This should be updated with the fix from https://github.com/01org/intel-arduino-tools/pull/58/commits/64df2c1f79a29db42ccba7fc27139ba5c44ab0e4

File: examples/Blink/Makefile [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
-include $(CODK_SW_DIR)/Makefile
+prog = Blink.ino
+ARDUINOSW_DIR ?=$(CODK_DIR)/arduino101_software
+
+SRCS    := $(wildcard *.c) \
+          $(wildcard *.cpp)
+OBJS1 := $(SRCS:.c=.o)
+OBJS  := $(OBJS1:.cpp=.o)
+VERBOSE = true
+include $(ARDUINOSW_DIR)/Makefile.inc
```
Comment (eriknyquist) [inline examples/Blink/Makefile]: Is there any way that lines 2-7 could go into the common Makefile.inc?

File: bin/x86/bin/install_script.sh (hunk 1 above)
Comment (calvinatintel) [inline bin/x86/bin/install_script.sh]: I don't think we need this file

Comment (bbaltz505) [inline bin/x86/bin/install_script.sh]: Yeah. this was checked in by mistake when I pulled over the sketch upload tools.

File: bin/parse_json.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+# parse JSON to search for Curie package/platform
+intel = (item for item in data["packages"] if item["name"] == "Intel").next()
+core = (item for item in intel["platforms"] if item["version"] == version and item["name"] == name).next()
+corelibs_url = core["url"]
+
+# Download corelibs URL
+print "Downloading Corelibs: %s" % corelibs_url
+r = requests.get(corelibs_url)
+open(local_zip, "wb").write(r.content)
```
Comment (eriknyquist) [inline bin/parse_json.py]: Should probably close the file, or use "with", i.e.
`with open(local_zip, "wb") as fh:`
`    fh.write(r.content)`

**Comment by bbaltz505** [comment]
@eriknyquist good point. I updated the script and removed install_script.sh

**Comment by bbaltz505** [comment]
@calvinatintel cleaned out some IDE stuff and fixed the corelibs install. Larry can at least build the C++ example

**Comment by bbaltz505** [comment]
this works on single-file sketches. debugging multi-file sketches

**Comment by bbaltz505** [comment]
make convert-sketch SKETCH=file.ino
...produces main.cpp

**Comment by eriknyquist** [comment]
Looks good @bbaltz505, I'll try it out later this evening and report back...

File: bin/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# arduino101load
+multiplatform launcher for Arduino101 dfu-util flashing utility
+
+## Compiling
+
+* Download go package from [here](https://golang.org/dl/) or using your package manager
+* `cd` into the root folder of this project
```
Comment (eriknyquist) [inline bin/README.md]: Well, it's not the root directory anymore... I had to cd into the "bin"directory to get the build to work

**Comment by eriknyquist** [comment]
@bbaltz505 I'm not actually sure how to use this... the README explains how to build arduino101load, but I can't get the sketch converter to work.

$ make convert-sketch SKETCH=test.ino`
`make: *** No rule to make target `convert-sketch'.  Stop`

**Comment by bbaltz505** [comment]
@calvinatintel this commit looks ok. Can you review? do you want to squash the commits and merge to master?

File: bin/99-dfu.rules [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Arduino 101 in DFU Mode
+SUBSYSTEM=="tty", ENV{ID_REVISION}=="8087", ENV{ID_MODEL_ID}=="0ab6", MODE="0666", ENV{ID_MM_DEVICE_IGNORE}="1", ENV{ID_MM_CANDIDATE}="0"
+SUBSYSTEM=="usb", ATTR{idVendor}=="8087", ATTR{idProduct}=="0aba", MODE="666", ENV{ID_MM_DEVICE_IGNORE}="1"
+<<<<<<< HEAD
```
Comment (calvinatintel) [inline bin/99-dfu.rules]: Prob a bad merge

File: examples/Blink/Blink.ino [hunk 6]
```diff
@@ -13,7 +13,10 @@
   by Scott Fitzgerald
  */
 
+<<<<<<< HEAD
```
Comment (calvinatintel) [inline examples/Blink/Blink.ino]: Prob a bad merge

File: libraries/README [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<<<<<<< HEAD
```
Comment (calvinatintel) [inline libraries/README]: Same. Merge problem

**Pull request merged**